connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufFunctionsSuite.scala

```diff
@@ -1248,6 +1248,23 @@ class ProtobufFunctionsSuite extends QueryTest with SharedSparkSession with Prot
     testFromProtobufWithOptions(df, expectedDfTwo, optionsTwo)
   }
 
+  test("Verify exceptions are correctly propagated with errors") {
+    // This triggers an query compilation error and ensures that original exception is
+    // also included in in the exception.
+
+    val invalidDescPath = "/non/existent/path.desc"
+
+    val ex = intercept[AnalysisException] {
```

Comment:
Shall we use `checkError` for checking the error?

Reply:
Added `checkError()`. Note that it does not replace the asserts since those assert check for 'getCause'.